ic-os/components/setupos-scripts/setup-hostos-config.sh

```diff
@@ -28,29 +30,61 @@ function copy_config_files() {
+
     echo "* Copying SSH authorized keys..."
-    if [ -d "${CONFIG_DIR}/ssh_authorized_keys" ]; then
-        cp -r ${CONFIG_DIR}/ssh_authorized_keys /media/
-        log_and_halt_installation_on_error "${?}" "Unable to copy SSH authorized keys to hostOS config partition."
+    use_ssh_authorized_keys=$(get_config_value '.icos_settings.use_ssh_authorized_keys')
+    if [[ "${use_ssh_authorized_keys,,}" == "true" ]]; then
+        if [ -d "${CONFIG_DIR}/ssh_authorized_keys" ]; then
+            cp -a "${CONFIG_DIR}/ssh_authorized_keys" /media/
+            log_and_halt_installation_on_error "${?}" "Unable to copy SSH authorized keys to hostOS config partition."
+        else
+            echo >&2 "Warning: SSH authorized keys are not configured."
+        fi
     else
-        log_and_halt_installation_on_error "1" "Directory 'ssh_authorized_keys' does not exist."
+        echo >&2 "SSH keys not in use."
     fi
 
     echo "* Copying node operator private key..."
-    if [ -f "${CONFIG_DIR}/node_operator_private_key.pem" ]; then
-        cp ${CONFIG_DIR}/node_operator_private_key.pem /media/
-        log_and_halt_installation_on_error "${?}" "Unable to copy node operator private key to hostOS config partition."
+    use_node_operator_private_key=$(get_config_value '.icos_settings.use_node_operator_private_key')
+    if [[ "${use_node_operator_private_key,,}" == "true" ]]; then
+        if [ -f "${CONFIG_DIR}/node_operator_private_key.pem" ]; then
+            cp "${CONFIG_DIR}/node_operator_private_key.pem" /media/
+            log_and_halt_installation_on_error "${?}" "Unable to copy node operator private key to hostOS config partition."
+        else
+            log_and_halt_installation_on_error "1" "use_node_operator_private_key set to true but not found"
+        fi
     else
-        echo "node_operator_private_key.pem does not exist, requiring HSM."
+        echo >&2 "Warning: node_operator_private_key.pem does not exist, requiring HSM."
     fi
 
-    echo "* Copying deployment.json to config partition..."
-    cp /data/deployment.json /media/
-    log_and_halt_installation_on_error "${?}" "Unable to copy deployment.json to hostOS config partition."
-
     echo "* Copying NNS public key to hostOS config partition..."
-    cp /data/nns_public_key.pem /media/
-    log_and_halt_installation_on_error "${?}" "Unable to copy NNS public key to hostOS config partition."
+    use_nns_public_key=$(get_config_value '.icos_settings.use_nns_public_key')
+    if [[ "${use_nns_public_key,,}" == "true" ]]; then
+        if [ -f "/data/nns_public_key.pem" ]; then
+            cp /data/nns_public_key.pem /media/
+            log_and_halt_installation_on_error "${?}" "Unable to copy NNS public key to hostOS config partition."
+        else
+            log_and_halt_installation_on_error "1" "use_nns_public_key set to true but not found."
+        fi
+    else
+        log_and_halt_installation_on_error "1" "use_nns_public_key must be set to true."
```

Comment:
Do I understand correctly that we always trigger this when use_nns_public_key is false? Why do we even allow to configure it if it must be true (just for backwards compatibility?)?

Reply:
Yes, currently, use_nns_public_key must be true for a successful install. The thought process was that later, when we introduce SEV, we will want to lock down using a custom NNS public key as a dev feature and hard-code the real key for production.

For context: https://dfinity.atlassian.net/browse/NODE-953